pkg/binding/controller.go

```diff
@@ -99,17 +100,28 @@ type Controller struct {
 func NewController(parentLogger logr.Logger, wdsRestConfig *rest.Config, imbsRestConfig *rest.Config,
 	wdsName string, allowedGroupsSet sets.Set[string]) (*Controller, error) {
 
-	dynamicClient, err := dynamic.NewForConfig(wdsRestConfig)
+	kubernetesClient, err := kubernetes.NewForConfig(wdsRestConfig)
 	if err != nil {
 		return nil, err
 	}
 
-	kubernetesClient, err := kubernetes.NewForConfig(wdsRestConfig)
+	extClient, err := apiextensionsclientset.NewForConfig(wdsRestConfig)
 	if err != nil {
 		return nil, err
 	}
 
-	extClient, err := apiextensionsclientset.NewForConfig(wdsRestConfig)
+	nGVRs, err := countGVRsViaAggregatedDiscovery(kubernetesClient)
+	if err != nil {
+		parentLogger.Error(err, "Not able to count GVRs via aggregated discovery") // but we can still continue
+	}
+	// tuning the rate limiter based on the number of GVRs is tested to be working well
+	wdsRestConfig.Burst = computeBurstFromNumGVRs(nGVRs)
+	wdsRestConfig.QPS = computeQPSFromNumGVRs(nGVRs)
+	parentLogger.V(1).Info("Parameters of the client's token bucket rate limiter", "burst", wdsRestConfig.Burst, "qps", wdsRestConfig.QPS)
```

Comment:
This logging should be done via the controller's logger rather than the parent logger.

Reply:
Controller, as the golang struct, doesn't have access to the values, right?

Reply:
I am not sure I understand. In the current version of this PR (https://github.com/kubestellar/kubestellar/blob/48f75f51b37bd325ad4c075aa15ad0c6b2158d6b/pkg/binding/controller.go), the controller does not even exist at the point in code that we are discussing. To use the controller's logger here would require moving the construction of that logger from inside `makeController` to somewhere earlier than the line we are discussing.

Reply:
> the controller does not even exist at the point in code that we are discussing

This is exactly what made me confused. So what you suggested is to construct the controller's log earlier and use it. I get the point. But what is the issue of using the `parentLogger` instead? BTW why is it called `parentLogger`, i.e. who is the child?

Reply:
The binding controller is just one of the controllers in the controller-manager process.